### PR TITLE
Snapshot tests

### LINKED ISSUE #12: Add CI and tests
- [x] simple snapshot tests: Read fixture, write out and compare for equality.
- [x] add CircleCI builds that run the tests

### PR DESCRIPTION
**Pull request by leowe**
- add snapshot tests with corresponding test files (pickled and unpickled)
- fix bug that caused program to crash when no group id was given
- closes #12

### CODE REVIEW

File: tests/test_utils.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import wknml
```
Comment (philippotto) [inline tests/test_utils.py]: I don't really understand the purpose of this file. What is the difference to `test_snapshot_readandwrite.py`? Due to the lodashes in the methods, it won't be picked up by pytest (that's why it has the `if __name__ ...` code?). However, it isn't executed on the CI?

Comment (leowe) [reply tests/test_utils.py]: The methods in this file are meant to be executed when there were changes in the behaviour of the nml parsing, so you can generate new (pickled/unpickled) files to compare against.

Comment (philippotto) [reply tests/test_utils.py]: Ah ok, then I'd suggest to rename the file to `refresh_snapshots.py`.

File: tests/test_utils.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+def __save_snapshot_pickle_read_and_write():
+    for i in range(0, len(INPUT_FILES)):
+        input_file = INPUT_FILES[i]
+        output_file = SNAPSHOT_FILES[i]
+        parsed = wknml.parse_nml(input_file)
+        with open(output_file + '.pickle', 'wb') as f:
+            pickle.dump(parsed, f)
+
+
+def __save_snapshot_nml_read_and_write():
```
Comment (philippotto) [inline tests/test_utils.py]: Why the `__` prefix?

Comment (leowe) [reply tests/test_utils.py]: so it doesn't get executed by pytest. Alternatively I could give pytest as a parameter the test file to test

Comment (leowe) [reply tests/test_utils.py]: Changed it.

Comment (philippotto) [reply tests/test_utils.py]: Ah, I thought pytest would only execute test functions which are prefixed with "test". Nevermind!

**Pull request merged**